src/migrator/utils.ts

```diff
@@ -215,6 +226,73 @@ WHERE "model name" = ${1}`,
 	}
 };
 
+export const lockMigrations = async <T>(
+	tx: Tx,
+	modelName: string,
+	blockingWait: boolean = true,
+	fn: () => Promise<T>,
+): Promise<T | undefined> => {
+	if (!(await migrationTablesExist(tx))) {
+		return;
+	}
+
+	let getLockStatement;
+
+	if (sbvrUtils.db.engine === Engines.websql) {
+		return $lockMigrations(tx, modelName, fn);
+	} else if (sbvrUtils.db.engine === Engines.mysql) {
+		// right now the mysql locks are not testable
+		// pinejs generates models that are not executable on mysql databases
+		return $lockMigrations(tx, modelName, fn);
+	} else if (sbvrUtils.db.engine === Engines.postgres) {
+		// pg_advisory lock expects an BigInt or two Int as lock identifier
+		// Therefore the model name is hashed and the first 8 bytes are taken as the Integer representation.
+		const modelKey: string = createHash('shake128', { outputLength: 8 })
+			.update('resin')
+			.digest()
+			.readBigInt64BE()
+			.toString();
+		getLockStatement = blockingWait
+			? `pg_advisory_xact_lock(${modelKey})`
+			: `pg_try_advisory_xact_lock(${modelKey})`;
```

Comment:
Can you use `tx.getTxLevelLock(...)` instead please, and extend it to allow for non-blocking operation as well

Reply:
Oh sure, thanks for the pointer